source/simulator/melt.cc

```diff
@@ -1202,11 +1188,35 @@ namespace aspect
     }
     prm.leave_subsection();
 
+    AssertThrow( !this->get_parameters().use_discontinuous_temperature_discretization &&
+                 !this->get_parameters().use_discontinuous_composition_discretization,
+                 ExcMessage("Melt transport can not be used with discontinuous elements.") );
+    AssertThrow( !this->get_parameters().free_surface_enabled,
+                 ExcMessage("Melt transport together with a free surface has not been tested.") );
+
     boundary_fluid_pressure.reset(BoundaryFluidPressure::create_boundary_fluid_pressure<dim>(prm));
+    if (SimulatorAccess<dim> *sim = dynamic_cast<SimulatorAccess<dim>*>(boundary_fluid_pressure.get()))
+      sim->initialize_simulator (this->get_simulator());
     boundary_fluid_pressure->parse_parameters (prm);
+    boundary_fluid_pressure->initialize ();
   }
 
 
+
+  template <int dim>
+  void
+  MeltHandler<dim>::
+  initialize ()
+  {
+    this->get_signals().set_assemblers.connect (std_cxx11::bind(&MeltHandler<dim>::set_melt_assemblers,
```

Comment:
I dislike that we rely on the signals for core functionality here. This will be problematic if a user wants to react to the assemblers but you can't know if the melt or the user signal will be triggered first. Can we maybe call ``set_melt_assemblers`` right before triggering this signal instead?